paasta_tools/paasta_remote_run.py

```diff
@@ -232,23 +232,26 @@ def build_executor_stack(
     service,
     instance,
     cluster,
+    role,
     pool,
     # TODO: move run_id into task identifier?
     run_id,
     system_paasta_config,
     framework_staging_timeout,
 ):
 
+    cluster_fqdn = system_paasta_config.get_cluster_fqdn_format().format(cluster=cluster)
     mesos_address = '{}:{}'.format(
-        mesos_tools.get_mesos_leader(), mesos_tools.MESOS_MASTER_PORT,
+        mesos_tools.find_mesos_leader("mesos." + cluster_fqdn),
```

Comment:
you don't need the mesos prefix - you'll get there just with paasta-{cluster}.yelp:MESOS_MASTER_PORT